src/libpriv/rpmostree-core.c

```diff
@@ -2394,25 +2398,39 @@ break_hardlinks_at (int             dfd,
   return TRUE;
 }
 
+typedef struct {
+  int tmpdir_dfd;
+  const char *name;
+  const char *evr;
+  const char *arch;
+} RelabelTaskData;
+
 static gboolean
-relabel_one_package (RpmOstreeContext *self,
-                     int               tmpdir_dfd,
-                     OstreeRepo     *repo,
-                     DnfPackage     *pkg,
-                     OstreeSePolicy *sepolicy,
-                     gboolean          *changed,
-                     GCancellable   *cancellable,
-                     GError        **error)
-{
-  g_autofree char *nevra = g_strdup (dnf_package_get_nevra (pkg));
+relabel_in_thread_impl (RpmOstreeContext *self,
+                        const char       *name,
+                        const char       *evr,
+                        const char       *arch,
+                        int               tmpdir_dfd,
+                        gboolean         *out_changed,
+                        GCancellable     *cancellable,
+                        GError          **error)
+{
+  if (g_cancellable_set_error_if_cancelled (cancellable, error))
```

Comment:
Don't we also need this in `import_in_thread` ? Otherwise, we won't even stop new imports, no?

Reply:
We check the cancellable each iteration of reading the archive, and `GTask` also checks the cancellable before actually running the thread; see `g_task_start_task_thread()`, so it's probably simplest to just drop the one in importing.

Reply:
Hmm no the `GTask` one is just if one uses `g_task_set_return_on_cancel()`, let's be conservative